src/components/triggersAndActions.ts

```diff
@@ -0,0 +1,176 @@
+import path from 'path';
+import getAppRootDir from './getAppRoot';
+
+const schedule = require('node-schedule');
+
+export interface DatabaseRecord {
+  [key: string]: any;
+  code: string;
+}
+
+export interface DatabaseResult {
+  rows: DatabaseRecord[];
+}
+
+export interface TriggerPayload {
+  id: number;
+  trigger: string;
+  table: string;
+  record_id: number;
+}
+
+export interface ActionPayload {
+  id: number;
+  code: string;
+  parameters: { [key: string]: any };
+}
+
+export interface Action {
+  code: string;
+  path: string;
+  name: string;
+  trigger: string;
+  condition: { [key: string]: any };
+  parameter_queries: { [key: string]: any };
+}
+
+const pluginFolder = path.join(getAppRootDir(), 'plugins'); //Change this to config file
+
+// Load actions from Database at server startup
+export const loadActions = async function (
+  client: { [key: string]: Function },
+  actionLibrary: { [key: string]: Function }
+) {
+  console.log('Loading Actions from Database...');
+
+  // const actionLibrary: { [key: string]: Function } = {};
+
+  client
+    .query('SELECT code, path, function_name FROM action_plugin')
+    .then((res: DatabaseResult) => {
+      res.rows.forEach((row) => {
+        const action = require(path.join(pluginFolder, row.path));
+        actionLibrary[row.code] = action[row.function_name];
+      });
+    });
+
+  console.log('Actions loaded');
+
+  // return actionLibrary;
+};
+
+// Load scheduled jobs from Database at server startup
+export const loadScheduledActions = async function (
+  client: { [key: string]: Function },
+  actionLibrary: { [key: string]: Function },
+  actionSchedule: any[]
+) {
+  // const actionSchedule: any[] = [];
+
+  console.log('Loading Scheduled jobs');
+  // Load from Database
+  client
+    .query(
+      `SELECT id, action_code, parameters, time_executed FROM action_queue WHERE status = 'Scheduled' ORDER BY time_executed`
+    )
+    .then((res: DatabaseResult) => {
+      res.rows.forEach((action) => {
+        // console.log(action);
+        const date = new Date(action.time_executed);
```

Comment:
Yes, you're right it is -- the only reason I didn't change it because I didn't want to change it everywhere (database, etc.) just yet. But yeah, makes sense to make it clearer -- I'll change it, but it's probably going to cause a merge conflict when merging my other branch with more substantial database changes #7 .

Originally the field was going to be just a log of when the Action had run, but then when thinking about scheduled actions, I realised it makes sense to use that value as the scheduled time as well, and just search for which ones are in the future at startup.